components/encoder/ams/ams_as5048.go

```diff
@@ -154,7 +149,7 @@ func makeAS5048Encoder(
 	if err := res.Reconfigure(ctx, deps, conf); err != nil {
 		return nil, err
 	}
-	if err := res.startPositionLoop(ctx); err != nil {
+	if err := res.startPositionLoop(ctx); err != nil { // initializes res.workers
```

Comment:
Although in practice `startPositionLoop` is only called once per struct, I don't like that a sloppy coder in the future could call it a second time and introduce all kinds of trouble (goroutines that cannot be stopped). I kinda want to inline the function so the future problems cannot happen, but also kinda think my concern is overblown and nobody is going to mess around with `startPositionLoop` in the future after all. What do you think?

Reply:
...I've spent more time wondering about this than it would take to change it. I'll just do it...